pandas/_libs/tslibs/timezones.pyx

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# cython: profile=False
+
+try: string_types = basestring
+except NameError: string_types = str
+
+
+cdef extern from "Python.h":
+    Py_ssize_t PY_SSIZE_T_MAX
+
+cdef extern from "datetime.h":
+    void PyDateTime_IMPORT()
+
+# import datetime C API
+PyDateTime_IMPORT
+
+
+import numpy as np
+cimport numpy as np
+from numpy cimport ndarray, int64_t, float64_t
+np.import_array()
+
+
+# dateutil compat
+from dateutil.tz import (tzoffset,
+                         tzlocal as _dateutil_tzlocal,
+                         tzfile as _dateutil_tzfile,
+                         tzutc as _dateutil_tzutc,
+                         tzstr as _dateutil_tzstr)
+
+import sys
+if sys.platform == 'win32' or sys.platform == 'cygwin':
+    # equiv pd.compat.is_platform_windows()
+    from dateutil.zoneinfo import gettz as _dateutil_gettz
+else:
+    from dateutil.tz import gettz as _dateutil_gettz
+
+
+from pytz.tzinfo import BaseTzInfo as _pytz_BaseTzInfo
+import pytz
+UTC = pytz.utc
+
+
+from cpython cimport PyBool_Check
+cdef extern from "numpy/ndarrayobject.h":
+    bint PyArray_IsIntegerScalar(object)
+
+cdef inline bint is_integer_object(object obj):
+    # Ported from util (which gets it from numpy_helper) to avoid
+    # direct dependency
+    return (not PyBool_Check(obj)) and PyArray_IsIntegerScalar(obj)
+
+
+cdef int64_t NPY_NAT = np.datetime64('nat').astype(np.int64)
+
+
+# TODO: Does this belong somewhere else?
```

Comment:
This PR predates #17289.  Specifying C dependencies in setup.py is a PITA and I wanted this module to be portable.  So I copied `total_seconds` instead of getting it from src/datetime_helper.h.

Regardless of whether #17289 is merged, this function can be removed.  Coming right up.